programs/mango-v4/src/instructions/perp_settle_fees.rs

```diff
@@ -43,18 +43,20 @@ pub fn perp_settle_fees(ctx: Context<PerpSettleFees>, max_settle_amount: u64) ->
     // Calculate PnL
     let pnl = perp_position.unsettled_pnl(&perp_market, oracle_price)?;
 
-    // Account perp position must have a loss to be able to settle against the fee account
-    require!(pnl.is_negative(), MangoError::ProfitabilityMismatch);
-    require!(
-        perp_market.fees_accrued.is_positive(),
-        MangoError::ProfitabilityMismatch
-    );
-
     let settleable_pnl = perp_position.apply_pnl_settle_limit(&perp_market, pnl);
-    require!(
-        settleable_pnl.is_negative(),
-        MangoError::ProfitabilityMismatch
-    );
+
+    if !pnl.is_negative() // Account perp position must have a loss to be able to settle against the fee account
+        || !perp_market.fees_accrued.is_positive()
+        || !settleable_pnl.is_negative()
```

Comment:
it's sufficient to exit on `!settleable_pnl.is_negative()` - settleable_pnl is always more restrictive than pnl.